### PR TITLE
Port grouping redo/v21

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
- https://redmine.openinfosecfoundation.org/issues/6795
- https://redmine.openinfosecfoundation.org/issues/6792

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1662

Previous PR: https://github.com/OISF/suricata/pull/10514

Changes since v20:
- use correct way to make sort deterministic

### CODE REVIEW

File: src/detect-engine-build.c [hunk 1]
```diff
@@ -1126,9 +1126,188 @@ static int RuleSetWhitelist(Signature *s)
...
+        BUG_ON(x->port > x->port2);
+        cnt++;
+    }
+    if (cnt <= 1)
+        return;
+
+    DetectPort **array = (DetectPort **)SCCalloc(cnt, sizeof(DetectPort *));
+    if (array == NULL)
+        return;
+
```
Comment (victorjulien) [inline src/detect-engine-build.c:1171]: comment: build temp array to sort with qsort. Assign a temporary id to resolve otherwise equal groups.

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.

ERROR: QA failed on SURI_TLPR1_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 137 | 144 | 105.11% |
| | SURI_TLPR1_stats_chk | | |
| .uptime | 754 | 804 | 106.63% | 

Pipeline 18783

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/10519

**Comment by inashivb** [comment]
This PR has been shared for testing with folks, so, I'm reopening it. Any following PRs are about cleanups and not core code changes.
If it deems fit to close this despite that, please feel free to close it again.

**Comment by inashivb** [comment]
The final PR (https://github.com/OISF/suricata/pull/10567) has now been merged and the testers have been updated of the same. So, closing this.

**Pull request closed without merging**